docs/contributing/contributing-code/contributing-code-tests/running-functional-tests.md

```diff
@@ -33,42 +33,72 @@ As much as possible, the tests use product functionality such as the Radius CLI
 
 ## Running the tests locally
 
+**Please note** that tests may be cached if you are running them a few times in a row. Please run `go clean -testcache` to make sure that the cache is invalidated before running a test.
+
 ### Prerequisites
 
 1. Place `rad` on your path
 2. Make sure `rad-bicep` is downloaded (`rad bicep download`)
 3. Make sure your [local dev environment is setup](../contributing-code-control-plane/running-controlplane-locally.md)
 4. Log into your Github account and [Generate PAT](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens)
 5. Log-in to the container registry of your Github organization.
-
-   `export CR_PAT=<your_pat>`
-
-   `echo $CR_PAT | docker login ghcr.io -u <your_username> --password-stdin`
-
+   1. `export CR_PAT=<your_pat>`
+   2. `echo $CR_PAT | docker login ghcr.io -u <your_username> --password-stdin`
 6. Publish Bicep test recipes by running `BICEP_RECIPE_REGISTRY=<registry-name> make publish-test-bicep-recipes`
 7. Publish Terraform test recipes by running `make publish-test-terraform-recipes`
 8. Change the visibility of the published packages to 'public'
 
-> ⚠️ The tests assume the Kubernetes namespace in use is `default`. If your environment is set up differently you will see
-> test failures.
-
+> ⚠️ The tests assume the Kubernetes namespace in use is `default`. If your environment is set up differently you will see test failures.
 > ⚠️ If you set environment variables for functional tests you may need to restart VS Code or other editors for them to take effect.
 
-### Run
+### Run Non-Cloud Functional Tests
+
+1. Required environment variables:
+   1. **TF_RECIPE_MODULE_SERVER_URL**: This is the URL for the Terraform Recipe Module Server. If you have run `make publish-test-terraform-recipes` you will see this URL at the end of that command.
```

Comment:
Running `make publish-test-terraform-recipes` is a prerequisite so let's make that explicit. Feel free to modify the suggested phrasing. Also, let's call it out in the prerequisite above to copy the output for later use.

```suggestion
   1. **TF_RECIPE_MODULE_SERVER_URL**: This is the URL for the Terraform Recipe Module Server. You can find this URL in the output of `make publish-test-terraform-recipes` from step 7 of prerequisite above.
```